relay_client/src/lib.rs

```diff
@@ -181,6 +179,46 @@ fn convert_subscription_result(
     }
 }
 
+/// Generate a random key for the `Sec-WebSocket-Key` header.
+pub fn generate_websocket_key() -> String {
+    // a base64-encoded (see Section 4 of [RFC4648]) value that,
+    // when decoded, is 16 bytes in length (RFC 6455)
+    let r: [u8; 16] = rand::random();
+    data_encoding::BASE64.encode(&r)
+}
+
+/// Converts a URL string into an HTTP request for initiating a WebSocket connection.
+fn into_client_request(url: &str) -> Result<HttpRequest<()>, RequestBuildError> {
+    let uri: Uri = url
+        .parse()
+        .map_err(|_| RequestBuildError::Url("Invalid url".to_owned()))?;
+    let authority = uri
+        .authority()
+        .ok_or(RequestBuildError::Url("Url has not authority".to_owned()))?
+        .as_str();
+    let host = authority
+        .find('@')
+        .map(|idx| authority.split_at(idx + 1).1)
+        .unwrap_or_else(|| authority);
+
+    // Check if the host is empty (excluding the port)
+    if host.split(':').next().unwrap_or("").is_empty() {
+        return Err(RequestBuildError::Url("EmptyHostName".to_owned()));
+    }
```

Comment:
```rust
    let host = authority.split('@').last().unwrap_or_default();

    // Check if the host is empty (excluding the port)
    if host.split(':').next().unwrap_or_default().is_empty() {
        return Err(RequestBuildError::Url("EmptyHostName".to_owned()));
    }
```
This code will work correctly for a url such as `wss://user:email@example.com@relay.walletconnect.com:443`